x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/telemetry/Transformable.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.telemetry;
+
+/**
+ * Provides an interface for converting an object into another form. This is to provide a way to convert
+ * the values of the CounterMap into a more serializable form that can be converted to json to send for telemetry.
+ * @param <T> The resulting type to convert into.
+ */
+public interface Transformable<T> {
```

Comment:
nit: Is the name `Transformable` from the telemetry package? If not, maybe we could use a different word like `convertable`. I just worry that `Transformable` sounds too much like Transforms. Just a thought.

Reply:
Na I just came up with that. `Convertable` sounds good to me!

Reply:
Could this be merged into the `Stats` interface by adding a `Writeable toSerialisableForm()` to `Stats`? If not can this class be package private at least

Reply:
👍 thanks Dave, will do.